modules/data.atmosphere/R/tdm_generate_subdaily_models.R

```diff
@@ -0,0 +1,226 @@
+# Begin Script
+#----------------------------------------------------------------------
+
+gen_subdaily_models <- function(outfolder, dat.train_file, in.prefix, tdf_file, 
+    n.beta, day.window, resids = F, parallel = F, n.cores = NULL, overwrite = TRUE, 
+    verbose = FALSE) {
+
+    pb.index <- 1
+    pb <- txtProgressBar(min = 1, max = 8, style = 3)
+    
+    # ----- 1.0 Read data & Make time stamps ---------- Load the data
+    
+    vars.info <- data.frame(CF.name = c("date", "year", "doy", "hour", 
+        "air_temperature", "precipitation_flux", "air_temperature_max", 
+        "air_temperature_min", "surface_downwelling_shortwave_flux_in_air", 
+        "surface_downwelling_longwave_flux_in_air", "air_pressure", "specific_humidity", 
+        "eastward_wind", "northward_wind", "wind_speed"))
+    dat.train <- list()
+    tem <- ncdf4::nc_open(dat.train_file)
+    dim <- tem$dim
+    for (j in seq_along(vars.info$CF.name)) {
+        if (exists(as.character(vars.info$CF.name[j]), tem$var) == FALSE) {
+            dat.train[[j]] <- NA
+        } else {
+            dat.train[[j]] <- ncdf4::ncvar_get(tem, as.character(vars.info$CF.name[j]))
+        }
+    }
+    names(dat.train) <- vars.info$CF.name
+    dat.train <- data.frame(dat.train)
+    
+    dat.train$date <- strptime(paste(dat.train$year, dat.train$doy + 1, 
+        dat.train$hour, sep = "-"), "%Y-%j-%H", tz = "GMT")
+    dat.train$time.hr <- as.numeric(difftime(dat.train$date, paste0((min(dat.train$year) - 
+        1), "-12-31 23:00:00"), tz = "GMT", units = "hour"))
+    dat.train$time.day <- as.numeric(difftime(dat.train$date, paste0((min(dat.train$year) - 
+        1), "-12-31 23:00:00"), tz = "GMT", units = "day")) - 1/24
+    dat.train$time.day2 <- as.integer(dat.train$time.day + 1/(48 * 2)) + 
+        1  # Offset by half a time step to get time stamps to line up
+    dat.train <- dat.train[order(dat.train$time.hr), ]
+    
+    
+    # ----- 1.1 Coming up with the daily means that are what we can
+    # use as predictors ----------
+    
+    train.day <- aggregate(dat.train[, c("air_temperature", "precipitation_flux", 
+        "surface_downwelling_shortwave_flux_in_air", "surface_downwelling_longwave_flux_in_air", 
+        "air_pressure", "specific_humidity", "wind_speed")], by = dat.train[, 
+        c("year", "doy")], FUN = mean)
+    names(train.day)[3:9] <- c("air_temperature_mean.day", "precipitation_flux.day", 
+        "surface_downwelling_shortwave_flux_in_air.day", "surface_downwelling_longwave_flux_in_air.day", 
+        "air_pressure.day", "specific_humidity.day", "wind_speed.day")
+    train.day$air_temperature_max.day <- aggregate(dat.train[, c("air_temperature")], 
+        by = dat.train[, c("year", "doy")], FUN = max)$x
+    train.day$air_temperature_min.day <- aggregate(dat.train[, c("air_temperature")], 
+        by = dat.train[, c("year", "doy")], FUN = min)$x
+    summary(train.day)
```

Comment:
If you feel the need to report this, I'd suggest sending it through logger.info since it's pretty non-essential.